objects/src/accounts/code.rs

```diff
@@ -56,27 +72,41 @@ impl AccountCode {
         self.method_tree.root().into()
     }
 
+    /// Returns a reference to the [ModuleAst] backing the [AccountCode].
+    pub fn module(&self) -> &ModuleAst {
+        &self.module
+    }
+
+    /// Returns a reference to the method tree.
+    pub fn method_tree(&self) -> &SimpleSmt {
+        &self.method_tree
+    }
+
     /// Returns the number of public interface methods defined for this account.
     pub fn num_methods(&self) -> usize {
-        todo!()
+        self.procedures.len()
     }
 
     /// Returns true if a method with the specified root is defined for this account.
-    pub fn has_method(&self, _root: Digest) -> bool {
-        todo!()
+    pub fn has_method(&self, root: Digest) -> bool {
+        self.procedures.binary_search_by(|r| r.cmp(&root)).is_ok()
     }
 
-    /// Returns an account interface method at the specified index.
+    /// Returns an method digest for the method with the specified index.
     ///
     /// # Panics
     /// Panics if the provided index is out of bounds.
-    pub fn get_method_by_index(&self, _index: usize) -> &Program {
```

Comment:
The initial interface was defined to return a `Program`.  I'm not sure if this makes sense and I have since refactored it to return the method `Digest`.